packages/page-pattern-modal/src/components/page-pattern-modal.tsx

```diff
@@ -153,7 +156,27 @@ class PagePatternModal extends Component< PagePatternModalProps, PagePatternModa
 		this.setState( { selectedCategory } );
 	};
 
-	closeModal = () => {
+	closeModal = ( event: CloseModalEvent ) => {
+		// As of Gutenberg 13.1, the editor will auto-focus on the title block
+		// automatically. See: https://github.com/WordPress/gutenberg/pull/40195.
+		// This ends up triggering a `blur` event on the Modal that causes it
+		// to close just after the editor loads. To circumvent this, we check if
+		// the `blur` event is related to the title auto-focus and if so,
+		// we ignore it so that the Modal stays open. It's important to note
+		// that this callback handles more than the  event, though. See the
+		// `CloseModalEvent` type for more info.
+
+		// Let's narrow-down the types to be able to safely handle the `blur` scenario
+		// `KeyboardEvent` doesn't have a `relatedTarget` property, and the `MouseEvent`'s
+		// `relatedTarget` type doesn't have a `className` property, though for those events
+		// the Modal can just close and we don't need the piece of logic below:
+		if ( 'relatedTarget' in event && event.relatedTarget && 'className' in event.relatedTarget ) {
+			if ( event.relatedTarget?.className?.match( /wp-block-post-title/ ) ) {
```

Comment:
Since we're using optional chaining, we can probably remove the outer if statement, as the inner if statement will only be true if those properties exist

Reply:
We can't. Typescript will complain that the `relatedTarget` and `className` properties don't exist in some of the unified types.

Reply:
To elaborate: The outer if satisfies TS (it's a[ type-guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types)), and the inner diff checks if the properties have a value, as they might be `null` at runtime.

Reply:
Ah, that's unfortunate. I'd expect it to functionally be the same, so I'm slightly surprised TS doesn't handle this use case. (Though after a search, it looks like there's an issue about it: #34974)